src/bvh/distance_traverse.rs

```diff
@@ -1,13 +1,36 @@
+use std::cmp::Ordering;
+use std::collections::BinaryHeap;
+
 use crate::aabb::Bounded;
 use crate::bounding_hierarchy::BHValue;
 use crate::bvh::{iter_initially_has_node, Bvh, BvhNode};
 use crate::ray::Ray;
 
 #[derive(Debug, Clone, Copy)]
-enum RestChild {
-    Left,
-    Right,
-    None,
+struct DistNodePair<T: PartialOrd> {
+    dist: T,
+    node_index: usize,
+}
+
+impl<T: PartialOrd> Eq for DistNodePair<T> {}
+
+impl<T: PartialOrd> PartialEq<Self> for DistNodePair<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.dist.eq(&other.dist)
+    }
+}
+
+#[allow(clippy::non_canonical_partial_ord_impl)]
+impl<T: PartialOrd> PartialOrd<Self> for DistNodePair<T> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.dist.partial_cmp(&other.dist)
+    }
+}
+
+impl<T: PartialOrd> Ord for DistNodePair<T> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.partial_cmp(other).unwrap()
+    }
```

Comment:
Let's avoid [the lint](https://rust-lang.github.io/rust-clippy/rust-1.84.0/index.html#non_canonical_partial_ord_impl) entirely by putting `self.dist.partial_cmp(&other.dist).unwrap()` in `Ord` and `Some(self.cmp(other))` in `PartialOrd`.